### ISSUE TITLE
Async validation strange behaviour

### ISSUE THREAD

**Issue by afitiskin** [erikras/redux-form#2056]
I had a very strange bug on my project when I used async validation for fields. During form filling everything works without any issues, but when I submit the form I got `Uncaught (in promise)` error. I checked every line of my code and found that if I resolve async validating promise with some data `redux-form` treat it as an error and fail form submission.

As I see, the problem is here:
```javascript
const handleErrors = rejected => errors => {
  if (errors && Object.keys(errors).length) {
    stop(errors)
    return errors
  } else if (rejected) {
    stop()
    throw new Error('Asynchronous validation promise was rejected without errors.')
  }
  stop()
  return Promise.resolve()
}
```
First `if` statements will not be not passed even if `rejected` is `false`. I don't know if it is a feature of a bug?

I forked the repo and added [example of the problem](https://github.com/afitiskin/redux-form/blob/broken-async-validation-example/examples/brokenAsyncValidation/src/asyncValidate.js#L10).
Form works until you submit it with valid values, you will get error in the console.

**Comment by gustavohenke** [comment]
Closing due to inactivity.

In case you still have problems, please open a formal bug report/feature request/docs improvement request.
Otherwise, please ask for help at StackOverflow.

**Comment by erikras** [comment]
Fix released in [`v7.4.0`](https://github.com/erikras/redux-form/releases/tag/v7.4.0).
